src/Services/Identity/Query/Application/UseCases/Events/ProjectUserDetailsWhenUserChangedInteractor.cs

```diff
@@ -0,0 +1,37 @@
+using Application.Abstractions;
+using Contracts.Services.Identity;
+
+namespace Application.UseCases.Events;
+
+public interface IProjectUserDetailsWhenUserChangedInteractor : 
+    IInteractor<DomainEvent.UserDeleted>,
+    IInteractor<DomainEvent.UserRegistered>,
+    IInteractor<DomainEvent.UserPasswordChanged> { }
+
+public class ProjectUserDetailsWhenUserChangedInteractor : IProjectUserDetailsWhenUserChangedInteractor
+{
+    private readonly IProjectionGateway<Projection.UserDetails> _projectionGateway;
+
+    public ProjectUserDetailsWhenUserChangedInteractor(IProjectionGateway<Projection.UserDetails> projectionGateway)
+    {
+        _projectionGateway = projectionGateway;
+    }
+
+    public Task InteractAsync(DomainEvent.UserDeleted @event, CancellationToken cancellationToken)
+        => _projectionGateway.DeleteAsync(@event.UserId, cancellationToken);
+
+    public Task InteractAsync(DomainEvent.UserRegistered @event, CancellationToken cancellationToken)
+    {
+        Projection.UserDetails userDetails = 
+            new(@event.UserId, @event.FirstName, @event.LastName, @event.Email, @event.Password, false);
+
+        return _projectionGateway.InsertAsync(userDetails, cancellationToken);
```

Comment:
Based on idempotency principles, it should be an Upsert. That is already adjusted on the release branch. Am I wrong?